### PR TITLE
Integrate the f8e8m0 with Constant and Convert

### PR DESCRIPTION
**Pull request by barnasm1**
### Details:
 * Add f16, bf16, f32 <-> f8e8m0 conversion in Convert operator
 * Add f8e8m0 support in Constant operator

### Tickets:
 - [*CVS-141563*](https://jira.devtools.intel.com/browse/CVS-141563)

### CODE REVIEW

File: src/core/src/op/convert.cpp [hunk 1]
```diff
@@ -232,6 +233,7 @@ bool Convert::has_evaluate() const {
         case element::u64:
         case element::f8e4m3:
         case element::f8e5m2:
+        case element::f8e8m0:
```
Comment (praasz) [inline src/core/src/op/convert.cpp:236]: Conversion to this float should be limited to  `f8e2m0` <-> `f16`, `bf16`, `f32`. 
Now the conversion will be generated for other types which will increase bin size.

File: src/bindings/python/src/openvino/runtime/opset13/ops.py [hunk 2]
```diff
@@ -137,7 +137,7 @@ def fake_convert(
                   of the same type as the data, and shape Numpy-broadcastable to data.
     :param shift: Optional tensor with value to subtract before and add after conversion of the data input value,
                   of the same type as the data, and shape Numpy-broadcastable to data.
-    :param destination_type: Type to emulate, string of either "f8e4m3" or "f8e5m2".
+    :param destination_type: Type to emulate, string of either "f8e4m3", "f8e5m2" or "f8e8m0".
```
Comment (praasz) [inline src/bindings/python/src/openvino/runtime/opset13/ops.py:140]: Is not required, fake convert has specified two types for support

File: src/common/transformations/src/transformations/common_optimizations/compress_float_constants.cpp [hunk 3]
```diff
@@ -96,7 +96,8 @@ class DetectCompressedWeights : public MatcherPass {
                                                                                            ov::element::u8,
                                                                                            ov::element::nf4,
                                                                                            ov::element::f8e4m3,
-                                                                                           ov::element::f8e5m2}));
+                                                                                           ov::element::f8e5m2,
+                                                                                           ov::element::f8e8m0}));
```
Comment (praasz) [inline src/common/transformations/src/transformations/common_optimizations/compress_float_constants.cpp:100]: The transformation can be skipped for now.
Can be expanded later if required.
@itikhono, could you confirm?

**Review by praasz — CHANGES_REQUESTED** [review]
Add python test to confirm usage Constant and/or Convert with new type can be used.
e.g. add test `src/bindings/python/tests/test_graph/test_constant.py`

File: src/bindings/python/src/openvino/runtime/opset13/ops.py (hunk 2 above)
Comment (barnasm1) [reply src/bindings/python/src/openvino/runtime/opset13/ops.py:140]: Done

File: src/common/transformations/src/transformations/common_optimizations/compress_float_constants.cpp (hunk 3 above)
Comment (barnasm1) [reply src/common/transformations/src/transformations/common_optimizations/compress_float_constants.cpp:100]: Done

File: src/core/src/op/convert_promote_types.cpp
[the remark this answers is not in the document]
Comment (barnasm1) [reply src/core/src/op/convert_promote_types.cpp:1]: Revert

File: src/core/src/op/convert.cpp (hunk 1 above)
Comment (barnasm1) [reply src/core/src/op/convert.cpp]: done

File: src/core/src/op/convert.cpp [hunk 4]
```diff
@@ -33,7 +34,8 @@ struct Evaluate : public element::NoAction<bool> {
     template <element::Type_t ET_IN,
```
Comment (praasz) [inline src/core/src/op/convert.cpp:34]: Update comment above regarding supported types for this conversion.

File: src/core/src/op/convert.cpp [hunk 5]
```diff
@@ -212,6 +229,10 @@ bool Convert::has_evaluate() const {
         return et == element::f16 || et == element::bf16 || et == element::f32 || et == element::f4e2m1;
     };
 
+    const auto can_convert_f8e8m0 = [](const element::Type& et) {
+        return et == element::f16 || et == element::bf16 || et == element::f32 || et == element::f8e8m0;
```
Comment (praasz) [inline src/core/src/op/convert.cpp:233]:
```suggestion
        et == element::f16 || et == element::bf16 || et == element::f32;
```
Can we make dedicated check for these types and use it in these lambdas?

**Review by praasz — APPROVED** [review]

File: src/core/src/op/convert.cpp (hunk 4 above)
Comment (barnasm1) [reply src/core/src/op/convert.cpp:34]: done

File: src/core/src/op/convert.cpp (hunk 5 above)
Comment (barnasm1) [reply src/core/src/op/convert.cpp]: Done

**Pull request merged**